### PR TITLE
Spread bulk emails through time

### PR DESCRIPTION
**Pull request by denys-medynskyi**
Resolves [github issue](https://github.com/pivorakmeetup/pivorak-web-app/issues/647)

### Issue

We can not send more than 100 emails per hour.

### Possible solutions
- switch to other mail service: **could not find free service** Any suggestions ?
- spread sending through time

### Description

This is simple hack that can help us with Mailgun issue.
I know it is not smart or bulletproof, but I think it is enough to make us safe from email errors.

All suggestions are appreciated.

### CODE REVIEW

**Comment by floor114** [comment]
@denysmedynskyi What about getting every n-th 100 records and delaying for n-th hours.

```ruby
event.visit_requests.approved.each_slice(100).with_index do |visit_requests, index|
  visit_requests.each do |visit_request|
    VisitRequestMailer.confirmation(visit_request).deliver_later(wait: index.hour)
  end
end
```

**Comment by denys-medynskyi** [comment]
@floor114 true, it can work. Let's go with your solution 👍

**Comment by denys-medynskyi** [comment]
@floor114 🙏 take a look. WDYT 🤔 ?

**Comment by floor114** [comment]
@denysmedynskyi LGTM. But one thing, I'd make some service that contains all split logic. Something like:

```ruby
SomeService.call(:confirmation_reminder, event.visit_requests.approved)

class SomeService
  def initialize(method_name, scope)
    ...
  end

  ....

  def call
    scope.each_slice(100).with_index do |objects, index|
      objects.each do |object|
        VisitRequestMailer.public_send(method_name, object).deliver_later(wait: index.hour)
      end
    end
  end
end
```

File: app/services/event/send_confirmation_reminders.rb [hunk 1]
```diff
@@ -1,13 +1,12 @@
 class Event
   class SendConfirmationReminders < ApplicationService
+
```
Comment (floor114) [inline app/services/event/send_confirmation_reminders.rb]: extra line

File: app/services/event/send_confirmations.rb [hunk 2]
```diff
@@ -1,13 +1,12 @@
 class Event
   class SendConfirmations < ApplicationService
+
```
Comment (floor114) [inline app/services/event/send_confirmations.rb]: extra line

**Review by floor114 — APPROVED** [review]

**Pull request merged**